.github/workflows/reusable-pre-commit.yml

```diff
@@ -0,0 +1,28 @@
+name: Pre-commit
+# The pre-commit configuration is in .pre-commit-config.yaml
+# we don't use the pre-commit action because it doesn't support local hooks in its virtual environment
+
+on:
+  workflow_call:
+    inputs:
+      ros_distro:
+        description: 'ROS2 distribution name'
+        required: true
+        type: string
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
```

Comment:
While writing my comment on `pip install` It came to my mind: Does it make sense to run setup-ros for any given distro on `ubuntu-latest`? Currently, all three distros run on ubuntu-22.04 but that will change for rolling in the near future.

Reply:
you are right, that's why I started adding `os_name` with #11. I'll add this here as well.